buildroot/share/PlatformIO/variants/MARLIN_F4x7Vx/variant.h

```diff
@@ -0,0 +1,192 @@
+#define PE7  71  // | 71      |            |            |                       |                      |                                   |           |
+#define PE8  72  // | 72      |            |            |                       |                      |                                   |           |
+#define PE9  73  // | 73      |            |            |                       |                      |                                   |           |
+#define PE10 74  // | 74      |            |            |                       |                      |                                   |           |
+#define PE11 75  // | 75      |            |            |                       |                      |                                   |           |
+#define PE12 76  // | 76      |            |            |                       |                      |                                   |           |
+#define PE13 77  // | 77      |            |            |                       |                      |                                   |           |
+#define PE14 78  // | 78      |            |            |                       |                      |                                   |           |
+#define PE15 79  // | 79      |            |            |                       |                      |                                   |           |
+//                  |---------|------------|------------|-----------------------|----------------------|-----------------------------------|-----------|
+#define PH0  80  // | 80      |            |            |                       |                      |                                   | OSC_IN    |
+#define PH1  81  // | 81      |            |            |                       |                      |                                   | OSC_OUT   |
+//                  |---------|------------|------------|-----------------------|----------------------|-----------------------------------|-----------|
+
+/// This must be a literal
+#define NUM_DIGITAL_PINS        82
+#define NUM_ANALOG_INPUTS       16
+
+// On-board LED pin number
+#ifndef LED_BUILTIN
+#define LED_BUILTIN             PA5
+#endif
+#define LED_GREEN               LED_BUILTIN
+
+// On-board user button
+#ifndef USER_BTN
+#define USER_BTN                PC13
+#endif
+
+// SPI definitions
+#define PIN_SPI_SS              PA4
+#define PIN_SPI_SS1             PA4
+#define PIN_SPI_SS2             PB12
+#define PIN_SPI_SS3             PA15
+#define PIN_SPI_MOSI            PA7
+#define PIN_SPI_MISO            PA6
+#define PIN_SPI_SCK             PA5
+
+// I2C definitions
+#define PIN_WIRE_SDA            PB9
+#define PIN_WIRE_SCL            PB8
+
+// Timer Definitions
+// Use TIM6/TIM7 when possible as servo and tone don't need GPIO output pin
+#define TIMER_TONE              TIM6
+#define TIMER_SERVO             TIM7
+
```

Comment:
You should also define `TIMER_SERIAL`, or else it will probably use TIM7 by default.

You could add it here, but if this will be used by more boards it might be better to define it in the pins file as an extra command line argument.

An example from the Rumba32:
`-DTIMER_SERIAL=TIM9`

Exactly which timer to use mostly depends on which pans happen to be needed for fan or other hardware PWM.

Reply:
Can I add here with a ifndef?

Reply:
You could. I wish every variant set timers inside ifndef. It’s less important inside Marlin variants since we can edit them when needed.